lake, media: add three benchmark docs with six diagrams (#23113)
pingcap/docs: tidb-cloud-lake/guides/benchmark-data-ingestion.md

---

# {{{ .lake }}} vs. Snowflake: Data Ingestion Benchmark

## Overview

We conducted four specific benchmarks to evaluate {{{ .lake }}} versus Snowflake:

1. **TPC-H SF100 Dataset Loading**: Focuses on loading performance and cost for a large-scale dataset (100GB, ~600 million rows).
2. **ClickBench Hits Dataset Loading**: Tests efficiency in loading a wide-table dataset (76GB, ~100 million rows, 105 columns), emphasizing challenges associated with high column counts.
3. **1-Second Freshness**: Measures the platforms' ability to ingest data within a strict 1-second freshness requirement.
4. **5-Second Freshness**: Compares the platforms' data ingestion capabilities under a 5-second freshness constraint.

## Platforms

- **[Snowflake](https://snowflake.com)**: A well-known cloud data platform emphasizing scalable compute, data sharing.
- **[{{{ .lake }}}](https://tidbcloud.com)**: A cloud-native data warehouse built on the open-source {{{ .lake }}} project, focusing on scalability and cost-efficiency.

## Benchmark Conditions

> Conducted on a `Small-Size` warehouse (16vCPU, AWS us-east-2) using data from the same S3 bucket.

Reviewer: ```suggestion
Conducted on a `Small-Size` warehouse using data from the same S3 bucket.

> **Note:**
>
> This comparison is based on Snowflake Gen1 standard warehouses (`GENERATION = '1'`).
```

## Performance and Cost Comparison

> - **TPC-H SF100 Data**: {{{ .lake }}} offers a **67% cost reduction** over Snowflake.

Author: https://github.com/pingcap/docs/pull/23113/files#r3451415676
```suggestion
- **TPC-H SF100 Data**: {{{ .lake }}} offers a **48% cost reduction** over Snowflake.
```

> - **ClickBench Hits Data**: {{{ .lake }}} achieves a **91% cost reduction**.

Author: Ref: https://github.com/pingcap/docs/pull/23113/files#r3451415683
```suggestion
- **ClickBench Hits Data**: {{{ .lake }}} achieves a **84% cost reduction**.
```

> - **1-Second Freshness**: {{{ .lake }}} loads **400 times** more data than Snowflake.

Reviewer: This cost reduction is still based on `$2/hour`. With Lake priced at `$3.2/hour`, 446s costs about `$0.40`, so the reduction vs Snowflake `$0.77` is about `48%`, not `67%`.

> - **5-Second Freshness**: {{{ .lake }}} loads over **27 times** more data.

Reviewer: This percentage also needs recalculation with `$3.2/hour`. 598s costs about `$0.53`; compared with Snowflake `$3.42`, the reduction is about `84%`, not `91%`.

## Data Ingestion Benchmarks

![Data loading benchmark](/media/tidb-cloud-lake/data-loading-benchmark.png)

### TPC-H SF100 Dataset

| Metric         | Snowflake | {{{ .lake }}} | Description               |
| -------------- | --------- | -------------- | ------------------------- |
| **Total Time** | 695s      | 446s           | Time to load the dataset. |

> | **Total Cost** | $0.77     | $0.25          | Cost of data loading.     |

Author: Ref: https://github.com/pingcap/docs/pull/23113/files#r3451415687
```suggestion
| **Total Cost** | $0.77     | $0.40          | Cost of data loading.     |
```

> - Data Volume: 100GB

Reviewer: Please recalculate this Lake cost with `$3.2/hour`: `446 / 3600 * 3.2 = $0.40` approximately.

- Rows: Approx. 600 million

### ClickBench Hits Dataset

| Metric         | Snowflake | {{{ .lake }}} | Description               |
| -------------- | --------- | -------------- | ------------------------- |
| **Total Time** | 51m 17s   | 9m 58s         | Time to load the dataset. |

> | **Total Cost** | $3.42     | $0.30          | Cost of data loading.     |

Author: Ref: https://github.com/pingcap/docs/pull/23113/files#r3451415691
```suggestion
| **Total Cost** | $3.42     | $0.53          | Cost of data loading.     |
```

> - Data Volume: 76GB

Reviewer: Please recalculate this Lake cost with `$3.2/hour`: `598 / 3600 * 3.2 = $0.53` approximately.

Assesses the volume of data that can be ingested within a 5-second freshness requirement.

| Metric         | Snowflake   | {{{ .lake }}} | Description                                     |
| -------------- | ----------- | -------------- | ----------------------------------------------- |
| **Total Time** | 5s          | 5s             | Loading time frame.                             |
| **Total Rows** | 90,000 Rows | 2,500,000 Rows | Volume of data successfully ingested within 5s. |

## Reproduce the Benchmark

You can reproduce the benchmark by following the steps below.

### Benchmark Environment

Both Snowflake and {{{ .lake }}} was tested under similar conditions:

| Parameter      | Snowflake                                                | {{{ .lake }}}                            |
| -------------- | -------------------------------------------------------- | ----------------------------------------- |
| Warehouse Size | Small                                                    | Small                                     |
| vCPU           | 16                                                       | 16                                        |

> | Price          | [$4/hour](https://www.snowflake.com/en/pricing-options/) | [$2/hour](https://www.pingcap.com/pricing/) |

Author: Ref: https://github.com/pingcap/docs/pull/23113/files#r3451415694
```suggestion
| Price          | [$4/hour](https://www.snowflake.com/en/pricing-options/) | [$3.2/hour](https://www.pingcap.com/pricing/) |
```

| AWS Region     | us-east-2                                                | us-east-2                                 |

> | Storage        | AWS S3                                                   | AWS S3                                    |

Reviewer: Lake pricing should be `$3.2/hour`, not `$2/hour`. Please update this row and recalculate the Lake cost and cost-reduction numbers on this page.

- The TPC-H SF100 dataset, sourced from [Amazon Redshift](https://github.com/awslabs/amazon-redshift-utils/tree/master/src/CloudDataWarehouseBenchmark/Cloud-DWB-Derived-from-TPCH).
- The ClickBench dataset, sourced from [ClickBench](https://github.com/ClickHouse/ClickBench).

### Prerequisites

- Have a [Snowflake account](https://signup.snowflake.com)
- Create a [{{{ .lake }}} account](https://tidbcloud.com/)

### Data Ingestion Benchmark

The data ingestion benchmark can be reproduced using the following steps:

<details>
  <summary>TPC-H Data Loading</summary>

1. **Snowflake Data Load**:

    - Log into your [Snowflake account](https://app.snowflake.com/).

>     - Create tables corresponding to the TPC-H schema. [SQL Script](https://github.com/databendlabs/benchmarks/blob/main/tpch-100/snowflake/setup.sql).

Reviewer: Please replace the Snowflake setup SQL links in this TPC-H data loading step with the hosted Lake asset: https://lakesql-bin.tidbcloud.com/datasets/tpch/tpch-100/snowflake/setup.sql

    - Use the `COPY INTO` command to load the data from AWS S3. [SQL Script](https://github.com/databendlabs/benchmarks/blob/main/tpch-100/snowflake/setup.sql).

2. **{{{ .lake }}} Data Load**:

    - Sign in to your [{{{ .lake }}} account](https://tidbcloud.com).

>     - Create the necessary tables as per the TPC-H schema. [SQL Script](https://github.com/databendlabs/benchmarks/blob/main/tpch-100/databend/setup.sql).

Reviewer: Please replace the Lake setup SQL links in this TPC-H data loading step with the hosted Lake asset: https://lakesql-bin.tidbcloud.com/datasets/tpch/tpch-100/lake/setup.sql

    - Use a method similar to Snowflake for loading data from AWS S3. [SQL Script](https://github.com/databendlabs/benchmarks/blob/main/tpch-100/databend/setup.sql).

</details>

<details>
  <summary>ClickBench Hits Data Loading</summary>

1. **Snowflake Data Load**:

    - Log into your [Snowflake account](https://app.snowflake.com/).

>     - Create tables corresponding to the `hits` schema. [SQL Script](https://gist.github.com/BohuTANG/2a23e5f829a8d180f7388c530526ab21?permalink_comment_id=4991762#file-hits-snowflake-schema).

Reviewer: Please replace this gist link with the hosted Snowflake schema SQL: https://lakesql-bin.tidbcloud.com/datasets/tpch/hits/snowflake/schema.sql

>     - Use the `COPY INTO` command to load the data from AWS S3. [SQL Script](https://gist.github.com/BohuTANG/2a23e5f829a8d180f7388c530526ab21?permalink_comment_id=4991762#gistcomment-4991762).

Reviewer: Please replace this gist comment link with the hosted Snowflake copy SQL: https://lakesql-bin.tidbcloud.com/datasets/tpch/hits/snowflake/copy.sql

2. **{{{ .lake }}} Data Load**:

    - Sign in to your [{{{ .lake }}} account](https://tidbcloud.com).

>     - Create the necessary tables as per the `hits` schema. [SQL Script](https://gist.github.com/BohuTANG/ab45d251c533dcf0b1ccd3ea1263b8a0#file-hits-databend-schema).

Reviewer: Please replace this gist link with the hosted Lake schema SQL: https://lakesql-bin.tidbcloud.com/datasets/tpch/hits/lake/schema.sql

>     - Use a method similar to Snowflake for loading data from AWS S3. [SQL Script](https://gist.github.com/BohuTANG/ab45d251c533dcf0b1ccd3ea1263b8a0?permalink_comment_id=4991767#gistcomment-4991767).

Reviewer: Please replace this gist comment link with the hosted Lake copy SQL: https://lakesql-bin.tidbcloud.com/datasets/tpch/hits/lake/copy.sql

---
title: "{{{ .lake }}} vs. Snowflake: Data Ingestion Benchmark"
summary: This page presents a benchmark comparison of data ingestion performance and cost between {{{ .lake }}} and Snowflake, focusing on TPC-H SF100 dataset loading, ClickBench Hits dataset loading, and freshness benchmarks.
---

# {{{ .lake }}} vs. Snowflake: Data Ingestion Benchmark

## Overview

We conducted four specific benchmarks to evaluate {{{ .lake }}} versus Snowflake:

> 1. **TPC-H SF100 Dataset Loading**: Focuses on loading performance and cost for a large-scale dataset (100GB, ~600 million rows).

Reviewer: Please double-check the TPC-H SF100 row count. If this refers to the full 8-table dataset, it is closer to ~875M rows based on the table row counts, not ~600M. If it only refers to `lineitem`, please say that explicitly and also align the later `Rows: Approx. 600 million` wording.

2. **ClickBench Hits Dataset Loading**: Tests efficiency in loading a wide-table dataset (76GB, ~100 million rows, 105 columns), emphasizing challenges associated with high column counts.
3. **1-Second Freshness**: Measures the platforms' ability to ingest data within a strict 1-second freshness requirement.
4. **5-Second Freshness**: Compares the platforms' data ingestion capabilities under a 5-second freshness constraint.

## Platforms

- **[Snowflake](https://snowflake.com)**: A well-known cloud data platform emphasizing scalable compute, data sharing.

> - **[{{{ .lake }}}](https://tidbcloud.com)**: A cloud-native data warehouse built on the open-source {{{ .lake }}} project, focusing on scalability and cost-efficiency.

Reviewer: `{{{ .lake }}}` is not an open-source project, so the "built on the open-source ... project" claim is inaccurate. Suggest removing it.

```suggestion
- **[{{{ .lake }}}](https://tidbcloud.com)**: A cloud-native data warehouse focusing on scalability and cost-efficiency.
```

| **Total Rows** | 100 Rows  | 40,000 Rows    | Volume of data successfully ingested within 1s. |

### 5-Second Freshness Benchmark

Assesses the volume of data that can be ingested within a 5-second freshness requirement.

| Metric         | Snowflake   | {{{ .lake }}} | Description                                     |
| -------------- | ----------- | -------------- | ----------------------------------------------- |
| **Total Time** | 5s          | 5s             | Loading time frame.                             |
| **Total Rows** | 90,000 Rows | 2,500,000 Rows | Volume of data successfully ingested within 5s. |

## Reproduce the Benchmark

You can reproduce the benchmark by following the steps below.

### Benchmark Environment

The benchmark tests both Snowflake and {{{ .lake }}} under similar conditions:

| Parameter      | Snowflake                                                | {{{ .lake }}}                            |
| -------------- | -------------------------------------------------------- | ----------------------------------------- |
| Warehouse Size | Small                                                    | Small                                     |

> | vCPU           | 16                                                       | 16                                        |

Reviewer: ```suggestion
```
